File: Linkedlist/js/linkedListFind.js
class Node{
    constructor(val, next= null){
        this.val = val
        this.next = next
    }
}

function linkedListFind(head, target){
    let curr = head

    while(curr !== null){
        if(curr.val === target) return true
        curr = curr.next
    }
    return false
}

function linkedListFind(head, target){
    if(head === null) return false

    if(head.next === target) return true

    return linkedListFind(head.next, target)
}

File: Linkedlist/python/linkedListFind.py
from __future__ import annotations
from dataclasses import dataclass
from typing import Optional


@dataclass
class Node:
    val: int
    next: Optional["Node"] = None


def linked_list_find(head: Optional[Node], target: int) -> bool:
    current = head
    while current is not None:
        if current.val == target:
            return True
        current = current.next
    return False
